pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -80,6 +83,80 @@ func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) f
 	return allErrs
 }
 
+type removedPlugins struct {
+	schemeGroupVersion string
+	plugins            []string
+}
+
+var removedPluginsByVersion = []removedPlugins{
+	{
+		schemeGroupVersion: v1beta1.SchemeGroupVersion.String(),
+		plugins:            []string{},
+	},
+	{
+		schemeGroupVersion: v1beta2.SchemeGroupVersion.String(),
+		plugins:            []string{"NodeLabel", "ServiceAffinity", "NodePreferAvoidPods"},
+	},
+}
+
+func isPluginRemoved(apiVersion string, name string) (bool, string) {
+	for _, dp := range removedPluginsByVersion {
+		for _, plugin := range dp.plugins {
+			if name == plugin {
+				return true, dp.schemeGroupVersion
+			}
+		}
+		if apiVersion == dp.schemeGroupVersion {
+			// We don't care about removals after this particular API version.
+			break
+		}
+	}
+	return false, ""
+}
+
+func validatePluginSetForRemovedPlugins(path *field.Path, apiVersion string, ps config.PluginSet) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i, plugin := range ps.Enabled {
+		if removed, removedVersion := isPluginRemoved(apiVersion, plugin.Name); removed {
+			allErrs = append(allErrs, field.Invalid(path.Child("enabled").Index(i), plugin.Name, fmt.Sprintf("was removed in version %q (KubeSchedulerConfiguration is version %q)", removedVersion, apiVersion)))
+		}
+	}
+	return allErrs
+}
+
+func validateProfilesForRemovedPlugins(path *field.Path, apiVersion string, profiles []config.KubeSchedulerProfile) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i, profile := range profiles {
```

Comment:
shouldn't this be part of `validateKubeSchedulerProfile` instead?

Reply:
Good point, moved it inside that function